spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcWriter.java

```diff
@@ -98,9 +106,9 @@ public SparkOrcValueWriter primitive(Type.PrimitiveType iPrimitive, TypeDescript
         case LONG:
           return SparkOrcValueWriters.longs();
         case FLOAT:
-          return SparkOrcValueWriters.floats();
+          return SparkOrcValueWriters.floats(getFieldId(primitive));
```

Comment:
nit: `getFieldId` is not used anywhere else, why not just use `ORCSchemaUtil.fieldId`